### PR TITLE
BI-13531 - Add terraform to deploy service to ECS

### PR DESCRIPTION
**Pull request by rmcmurray-ch**
Fulfils [BI-13531](https://companieshouse.atlassian.net/browse/BI-13531)

[BI-13531]: https://companieshouse.atlassian.net/browse/BI-13531?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: terraform/groups/ecs-service/locals.tf [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+locals {
+  stack_name                 = "search-service" # this must match the stack name the service deploys into
+  name_prefix                = "${local.stack_name}-${var.environment}"
+  global_prefix              = "global-${var.environment}"
+  service_name               = "alphabetical-company-search-consumer"
+  container_port             = 8080
+  docker_repo                = "alphabetical-company-search-consumer"
+  kms_alias                  = "alias/${var.aws_profile}/environment-services-kms"
+  lb_listener_rule_priority  = 21
+  lb_listener_paths          = ["/verification/*"]
```
Comment (VaughanJackson123) [inline terraform/groups/ecs-service/locals.tf]: * I suspect `lb_listener_paths` may be empty in this case, as the consumer's only REST handling will be for the health check?
* For the health check, we may want to add `healthcheck_path` and `healthcheck_matcher` variables?

File: terraform/groups/ecs-service/main.tf [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    }
+    vault = {
+      source  = "hashicorp/vault"
+      version = "~> 3.18.0"
+    }
+  }
+}
+
+module "secrets" {
+  source = "git@github.com:companieshouse/terraform-modules//aws/ecs/secrets?ref=1.0.216"
```
Comment (VaughanJackson123) [inline terraform/groups/ecs-service/main.tf]: Is there a specific reason why we cannot use the latest version of `terraform-modules` `1.0.245`?

File: terraform/groups/ecs-service/main.tf [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  vpc_id                  = data.aws_vpc.vpc.id
+  ecs_cluster_id          = data.aws_ecs_cluster.ecs_cluster.id
+  task_execution_role_arn = data.aws_iam_role.ecs_cluster_iam_role.arn
+
+  # Load balancer configuration
+  lb_listener_arn           = data.aws_lb_listener.service_lb_listener.arn
+  lb_listener_rule_priority = local.lb_listener_rule_priority
+  lb_listener_paths         = local.lb_listener_paths
+  health_check_grace_period_seconds = 240
+  healthcheck_healthy_threshold     = "2"
```
Comment (VaughanJackson123) [inline terraform/groups/ecs-service/main.tf]: Maybe move these down together with the other health check variables?

File: terraform/groups/ecs-service/profiles/development-eu-west-2/cidev/vars [hunk 4]
```diff
@@ -0,0 +1,4 @@
+environment = "cidev"
+aws_profile = "development-eu-west-2"
+
+log_level = "debug"
```
Comment (VaughanJackson123) [inline terraform/groups/ecs-service/profiles/development-eu-west-2/cidev/vars]: Is there any reason why `use_set_environment_files` has been omitted?

File: terraform/groups/ecs-service/vault.tf [hunk 5]
```diff
@@ -0,0 +1 @@
+ln -s vault-providers/userpass vault.tf
```
Comment (VaughanJackson123) [inline terraform/groups/ecs-service/vault.tf]: Is this actually a symbolic link? Compare with https://github.com/companieshouse/certificates.orders.web.ch.gov.uk/blob/master/terraform/groups/ecs-service/vault.tf and https://github.com/companieshouse/identity-verification-api/blob/main/terraform/groups/ecs-service/vault.tf?

**Comment by rmcmurray-ch** [comment]
@VaughanJackson123 addressed the PR comments in latest commit, if you have a chance for re-review, that'd be great thank you! 👍

File: terraform/groups/ecs-service/vault.tf [hunk 6]
```diff
@@ -0,0 +1 @@
+./vault-providers/userpass
```
Comment (VaughanJackson123) [inline terraform/groups/ecs-service/vault.tf]: This still doesn't appear to be a symbolic link?

**Comment by VaughanJackson123** [comment]
> @VaughanJackson123 addressed the PR comments in latest commit, if you have a chance for re-review, that'd be great thank you! 👍

I have taken another look @rmcmurray-ch. There are a couple of outstanding points still.

File: terraform/groups/ecs-service/main.tf (hunk 2 above)
Comment (rmcmurray-ch) [reply terraform/groups/ecs-service/main.tf]: oops did update this but must have not saved before push - apologies will catch it in the next push 👍

File: terraform/groups/ecs-service/vault.tf (hunk 6 above)
Comment (rmcmurray-ch) [reply terraform/groups/ecs-service/vault.tf]: sorry could you possibly elaborate? I went off the examples in the above links you gave, but I don't know what exactly makes it a symbolic link so didn't realise it was not one. Thanks!

Comment (VaughanJackson123) [reply terraform/groups/ecs-service/vault.tf]: I think the idea is that the file `vault.tf` can be/should be a symbolic link to the file `./vault-providers/userpass`.

**Review by VaughanJackson123 — APPROVED** [review]

**Pull request merged**